### PR TITLE
Create metadata bundle for openshift scanner

### PR DESCRIPTION
**Pull request by darkowlzz**
This adds a make target to generate metadata bundle and adds a test for
the generated bundle in the e2e test.
The bundle includes a separate CSV file with container image from rhel
repository and some updated attributes.

Also adds `deploy/olm/storageos/storageos.clusterserviceversion.yaml` to
maintain the latest version of CSV separately, not modifying the
published CSV.

### CODE REVIEW

File: deploy/olm/csv-rhel/storageos.v1.0.0.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+  namespace: placeholder
+  annotations:
+    capabilities: Full Lifecycle
+    categories: "Storage"
+    description: Cloud-native, persistent storage for containers.
+    containerImage: registry.connect.redhat.com/storageos/cluster-operator:1.0.0
+    repository: https://github.com/storageos/cluster-operator
+    createdAt: 2019-03-25T08:00:00Z
+    support: StorageOS, Inc
+    certified: "true"
```
Comment (darkowlzz) [inline deploy/olm/csv-rhel/storageos.v1.0.0.clusterserviceversion.yaml]: new

File: deploy/olm/csv-rhel/storageos.v1.0.0.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+    Make sure to install the Security Context Constraint (SCC) in order for
+    the StorageOS DaemonSet and StatefulSet pods to start properly:
+
+    ``` $ oc adm policy add-scc-to-user privileged system:serviceaccount:<namespace>:storageos-daemonset-sa ```
+
+    ``` $ oc adm policy add-scc-to-user privileged system:serviceaccount:<namespace>:storageos-statefulset-sa ```
+
+    By default, the StorageOS cluster is deployed in `storageos` namespace. The
+    service accounts in the above commands should be
+    `:storageos:storageos-daemonset-sa` and `:storageos:storageos-statefulset-sa`.
```
Comment (darkowlzz) [inline deploy/olm/csv-rhel/storageos.v1.0.0.clusterserviceversion.yaml]: SCC requirements for openshift.

Comment (darkowlzz) [reply deploy/olm/csv-rhel/storageos.v1.0.0.clusterserviceversion.yaml]: Maybe we should add this to the upstream k8s CSV as well and mention that this is a requirement for openshift only.

File: deploy/olm/csv-rhel/storageos.clusterserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+  namespace: placeholder
+  annotations:
+    capabilities: Full Lifecycle
+    categories: "Storage"
+    description: Cloud-native, persistent storage for containers.
+    containerImage: registry.connect.redhat.com/storageos/cluster-operator:1.0.0
+    repository: https://github.com/storageos/cluster-operator
+    createdAt: 2019-03-25T08:00:00Z
+    support: StorageOS, Inc
+    certified: "true"
```
Comment (darkowlzz) [inline deploy/olm/csv-rhel/storageos.clusterserviceversion.yaml]: Updated.

File: deploy/olm/csv-rhel/storageos.clusterserviceversion.yaml [hunk 4]
```diff
@@ -0,0 +1,355 @@
...
+    On OpenShift, make sure to install the Security Context Constraint (SCC) in
+    order for the StorageOS DaemonSet and StatefulSet pods to start properly:
+
+    ``` $ oc adm policy add-scc-to-user privileged system:serviceaccount:<namespace>:storageos-daemonset-sa ```
+
+    ``` $ oc adm policy add-scc-to-user privileged system:serviceaccount:<namespace>:storageos-statefulset-sa ```
+
+    By default, the StorageOS cluster is deployed in `storageos` namespace. The
+    service accounts in the above commands should be
+    `:storageos:storageos-daemonset-sa` and `:storageos:storageos-statefulset-sa`.
```
Comment (darkowlzz) [inline deploy/olm/csv-rhel/storageos.clusterserviceversion.yaml]: Additional info for openshift.

File: deploy/olm/csv-rhel/storageos.clusterserviceversion.yaml [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+    mediatype: image/png
```
Comment (croomes) [inline deploy/olm/csv-rhel/storageos.clusterserviceversion.yaml]: For the icon, can you use the bee logo as svg instead?  This should work:
```
icon:
  - base64data: PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0idXRmLTgiPz4NCjwhLS0gR2VuZXJhdG9yOiBBZG9iZSBJbGx1c3RyYXRvciAxOS4wLjAsIFNWRyBFeHBvcnQgUGx1Zy1JbiAuIFNWRyBWZXJzaW9uOiA2LjAwIEJ1aWxkIDApICAtLT4NCjxzdmcgdmVyc2lvbj0iMS4xIiBpZD0iTGF5ZXJfMSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgeD0iMHB4IiB5PSIwcHgiDQoJIHZpZXdCb3g9Ii05MCAyODQuNyA0MzAgNDMwIiBlbmFibGUtYmFja2dyb3VuZD0ibmV3IC05MCAyODQuNyA0MzAgNDMwIiB4bWw6c3BhY2U9InByZXNlcnZlIj4NCjxnPg0KCTxwYXRoIGZpbGw9IiM0RjUyNjMiIGQ9Ik0yNzguNSw0MjAuOWgtNzQuMWwtNS42LTkuN2wxMy42LTIzLjZsLTE4LjEtMzEuM0gxODBsMjcuNS00Ny43YzMtNS4xLDEuMi0xMS43LTMuOS0xNC42DQoJCWMtNS4xLTMtMTEuNy0xLjItMTQuNiwzLjlsLTM0LDU4LjhIOTMuMmwtMzQtNTguOGMtMi45LTUuMS05LjUtNi45LTE0LjYtMy45cy02LjksOS41LTMuOSwxNC42bDI3LjUsNDcuN0g1My45bC0xOC4xLDMxLjMNCgkJbDEzLjYsMjMuNmwtNS42LDkuOGgtNzQuMmwtNDMuMyw3NC45bDQzLjMsNzQuOWg3NC4xbDM0LDU4LjlMNzksNjI5bC0xLjIsMC44bDQ2LjQsNzcuMmw0Ni4yLTc3LjNsMCwwbDM0LTU4LjloNzQuMWw0My4zLTc0LjkNCgkJTDI3OC41LDQyMC45eiBNNTUuNSw0MDAuNWwtNy40LTEyLjhMNjAsMzY3aDE0LjNsMC4zLDAuNUw1NS41LDQwMC41eiBNMTczLjQsMzY3LjVsMC4zLTAuNUgxODhsMTEuOSwyMC42bC03LjQsMTIuOUwxNzMuNCwzNjcuNXoiDQoJCS8+DQoJPHBvbHlnb24gZmlsbD0iI0ZGRkZGRiIgcG9pbnRzPSI0My43LDQ0Mi4zIDQyLjgsNDQyLjMgLTE4LjEsNDQyLjMgLTQ4LjksNDk1LjggLTE4LjEsNTQ5LjMgNDMsNTQ5LjMgNDMuNyw1NDkuMyA3NC42LDQ5NS45IAkiLz4NCgk8cG9seWdvbiBmaWxsPSIjRkZGRkZGIiBwb2ludHM9IjI2Ni4xLDQ0Mi4zIDIwNC4zLDQ0Mi4zIDIwNC4zLDQ0Mi41IDE3My40LDQ5NS44IDIwNC4zLDU0OS4zIDIwNC40LDU0OS4zIDI2Ni4xLDU0OS4zIDI5Nyw0OTUuOCAJDQoJCSIvPg0KCTxwb2x5Z29uIGZpbGw9IiM2MUMyMDIiIHBvaW50cz0iMTU0LjksMzc4LjIgOTMuMSwzNzguMiA5My4xLDM3OC4zIDYyLjMsNDMxLjcgOTMuMSw0ODUuMiA5My4yLDQ4NS4yIDE1NC45LDQ4NS4yIDE4NS44LDQzMS43IAkiLz4NCgk8cG9seWdvbiBmaWxsPSIjNjFDMjAyIiBwb2ludHM9IjE1Mi4xLDYxOC40IDE1Mi4xLDYxOC40IDEyNCw2NjUuMiA5Ni4yLDYxOC45IDk2LjIsNjE4LjkgNjIuMiw1NjAuMSA4OC4zLDUxNC45IDkzLjEsNTA2LjYgDQoJCTE1NC45LDUwNi42IDE4NS44LDU2MC4xIAkiLz4NCjwvZz4NCjxyZWN0IHg9Ii0xMDUiIHk9IjI3MC43IiBmaWxsPSJub25lIiB3aWR0aD0iNDU4IiBoZWlnaHQ9IjQ1OCIvPg0KPC9zdmc+DQo=
    mediatype: image/svg
```

Comment (croomes) [reply deploy/olm/csv-rhel/storageos.clusterserviceversion.yaml]: btw, icon change should be applied to Operator Hub CSV too (I can do separately if you want)

Comment (darkowlzz) [reply deploy/olm/csv-rhel/storageos.clusterserviceversion.yaml]: I'll do it as part of this PR itself.

Comment (darkowlzz) [reply deploy/olm/csv-rhel/storageos.clusterserviceversion.yaml]: `spec.icon.mediatype in body should be one of [image/gif image/jpeg image/png image/svg+xml]
` 😁

Comment (darkowlzz) [reply deploy/olm/csv-rhel/storageos.clusterserviceversion.yaml]: image

**Pull request merged**